README.md

```diff
@@ -0,0 +1,258 @@
+
+# Composer is needed to install drush on the server.
+require 'capistrano/composer'
+
+# Drupal Tasks.
+require 'capdrupal/capistrano'
+
+# Load custom tasks from `lib/capistrano/tasks` if you have any defined.
+Dir.glob('config/capistrano/tasks/*.rake').each { |r| import r }
+```
+
+Then, go to `config/deploy.rb` to set the parameters of your project. First you have to define the general information about the user, server and the app himself.
+
+```ruby
+set :application, 'application-name'
+set :repo_url, 'git@github.com:company/application.git'
+
+server 'ssh.example.org', user: 'username', roles: %w{app db web}
+```
+
+The specific Drupal information
+
+```ruby
+set :install_composer, true
+set :install_drush, true
+
+set :app_path, 'web'
+set :config_path, 'config/d8/sync'
+
+# Setup the backup before/after failed strategy.
+set :backup_path, 'backups'
+set :keep_backups, 5
+
+# Link file settings.php
+set :linked_files, fetch(:linked_files, []).push("#{fetch(:app_path)}/sites/default/settings.php", "drush/drush.yml")
+
+# Link dirs files and private-files
+set :linked_dirs, fetch(:linked_dirs, []).push("#{fetch(:app_path)}/sites/default/files")
+```
+
+Then, all the others information related to your Git repository or debug level
+
+```ruby
+# Default value for :scm is :git
+set :scm, :git
+
+# Default value for :log_level is :debug
+set :log_level, :debug
+```
+
+Finally, set the deployment to use the proper Drupal 8 strategy
+
+```ruby
+namespace :deploy do
+  # Ensure everything is ready to deploy.
+  after "deploy:check:directories", "drupal:db:backup:check"
+
+  # Backup the database before starting a deployment and rollback on fail.
+  before :starting, "drupal:db:backup"
+  before :failed, "drupal:db:rollback"
+
+  # Set the maintenance Mode on your Drupal online project when deploying.
+  after :updated, "drupal:maintenance:on"
+
+  # Must updatedb before import configurations, E.g. when composer install new
+  # version of Drupal and need updatedb scheme before importing new config.
+  # This is executed without raise on error, because sometimes we need to do drush config-import before updatedb.
+  after :updated, "drupal:updatedb:silence"
+
+  # Remove the cache after the database update
+  after :updated, "drupal:cache:clear"
+  after :updated, "drupal:config:import"
+
+  # Update the database after configurations has been imported.
+  after :updated, "drupal:updatedb"
+
+  # Clear your Drupal 8 cache.
+  after :updated, "drupal:cache:clear"
+
+  # Disable the maintence on the Drupal project.
+  after :updated, "drupal:maintenance:off"
+
+  # Ensure permissions are properly set.
+  after :updated, "drupal:permissions:recommended"
+  after :updated, "drupal:permissions:writable_shared"
+
+  # Cleanup old database backups
+  before :cleanup, "drupal:db:backup:cleanup"
+
+  # Fix the release permissions (due to Drupal restrictive permissions)
+  # before deletting old release.
+  before :cleanup, :fix_permission do
```

Comment:
see #28 to move this into his own tasks (or do it automatically if we choosed a permissions strategy). This is linked to the task `drupal:permissions:recommended`

Reply:
what about that?

Reply:
create a task `drupal:permissions:cleanup` ?